### ISSUE TITLE
Real-world example for non-spatial geoprocessing

### ISSUE THREAD

**Issue by enauj** [GIScience/gis-training-resource-center#119]
In [Module 5: Non-Spatial Geoprocessing](https://giscience.github.io/gis-training-resource-center/content/Modul_5/en_qgis_non_spatial_tools.html) it would be really nice to have one or two examples about calculating a new field in the attribute table (using the field calculator) and representing the new field on a map. 
Could also be merging two datasets non-spatially or calculating an indicator.

**Comment by ioalexei** [comment]
For merging two datasets, an obvious one would be merging population data with admin boundaries based on p-codes (both from OCHA Common Operational Datasets published on HDX) - this is very common.

Another very common task is using field calculator to cast data type where it's been encoded or imported incorrectly, e.g. from string to int. This one isn't specific to humanitarian uses/data but is something people will come across a lot using messy data.

**Comment by ioalexei** [comment]
Another example that could combine field calculator and attribute join: 
if you have data only with e.g. admin3 pcode but boundaries only at admin2, you can calculate a field to take just the admin2 part of the admin3 pcode and join that to an admin 2 layer... 

e.g. if admin 3 pcode is 010204, you can derive admin 2 pcode with formula left($admin3, 4)

**Comment by isikl** [comment]
> For merging two datasets, an obvious one would be merging population data with admin boundaries based on p-codes (both from OCHA Common Operational Datasets published on HDX) - this is very common.

Yes I agree! This is often very much needed. Another similiar example: Joining admin boundaries with data on Multi Poverty Index (available in Excel file) containing pcodes. This exercise could be reused: https://giscience.github.io/gis-training-resource-center/content/Modul_5/en_qgis_modul_5_ex3.html#task-3-mpi-data but there might be definetly more adequate examples. reach out to me or Alec if you have question to this exercise. 

> Another very common task is using field calculator to cast data type where it's been encoded or imported incorrectly, e.g. from string to int. This one isn't specific to humanitarian uses/data but is something people will come across a lot using messy data.

In the newer QGIS versions there is now a tool called "Refactor Fields", which might replace this task in the future. However, practicing this step is always good and an easy example.
